test/test_transforms_v2_refactored.py

```diff
@@ -228,25 +228,46 @@ def check_functional_kernel_signature_match(functional, *, kernel, input_type):
         assert functional_param == kernel_param
 
 
-def _check_transform_v1_compatibility(transform, input):
+def _check_transform_v1_compatibility(v2_transform_eager, *inputs, rtol, atol):
     """If the transform defines the ``_v1_transform_cls`` attribute, checks if the transform has a public, static
-    ``get_params`` method, is scriptable, and the scripted version can be called without error."""
-    if transform._v1_transform_cls is None:
+    ``get_params`` method that is the v1 equivalent, and the output is close to v1 in eager and scripted mode."""
+    if len(inputs) != 1:
         return
 
-    if type(input) is not torch.Tensor:
+    input = inputs[0]
+    if type(input) is not torch.Tensor or isinstance(input, PIL.Image.Image):
         return
 
-    if hasattr(transform._v1_transform_cls, "get_params"):
-        assert type(transform).get_params is transform._v1_transform_cls.get_params
+    v1_transform_cls = v2_transform_eager._v1_transform_cls
+    if v1_transform_cls is None:
+        return
 
-    scripted_transform = _script(transform)
-    with ignore_jit_no_profile_information_warning():
-        scripted_transform(input)
+    if hasattr(v1_transform_cls, "get_params"):
+        assert type(v2_transform_eager).get_params is v1_transform_cls.get_params
+
+    v1_transform_eager = v1_transform_cls(**v2_transform_eager._extract_params_for_v1_transform())
+
+    def check_close(transform_v2, transform_v1):
+        torch.manual_seed(0)
+        output_v2 = transform_v2(input)
+
+        torch.manual_seed(0)
+        output_v1 = transform_v1(input)
```

Comment:
For both of these can we use a CM that makes sure the RNG isn't leaked beyond those calls? We probably have something for that already, otherwise we can do with https://pytorch.org/docs/stable/random.html#torch.random.fork_rng

I know we restore the RNG on a test-basis with our fixture, but I want to avoid having to think beyond those calls